vercel/resource_project.go

```diff
@@ -1101,6 +1111,11 @@ func (r *ResourceConfig) toClientResourceConfig(onDemandConcurrentBuilds types.B
 	if !onDemandConcurrentBuilds.IsUnknown() {
 		resourceConfig.ElasticConcurrencyEnabled = onDemandConcurrentBuilds.ValueBoolPointer()
 	}
+	if !buildMachineType.IsUnknown() && *resourceConfig.ElasticConcurrencyEnabled {
```

Comment:
What if `resourceConfig.ElasticConcurrencyEnabled` is `null`?

Reply:
I don't think we need this logic here, we can just use a validator, and set `buildMachineType` without worrying about its relationship to elastic-concurrency.